rust/template/src/main.rs

```diff
@@ -205,8 +205,15 @@ pub fn main() {
     fn no_op(_table: usize, _rec: &Record, _w: isize) {}
     let cb = if args.print { record_upd } else { no_op };
 
-    let hddlog = HDDlog::run(args.workers, args.store, cb);
+    match HDDlog::run(args.workers, args.store, cb) {
+        Ok(hddlog) => run(hddlog, args.delta),
+        Err(err) => {
+            eprintln!("Failed to run differential datalog: {}", err);
+            1
+        }
+    }
+}
 
-    let ret = run(hddlog, args.delta);
-    exit(ret);
+pub fn main() {
```

Comment:
Can we use this trick from Rust 2018?
```
// Rust 2018
use std::fs::File;

fn main() -> Result<(), std::io::Error> {
    let f = File::open("bar.txt")?;

    Ok(())
}
```

Reply:
Yeah, I suppose we can. I tried to stay away from it because based on my own experience (in at least three small projects) this API in its current form is insufficient in so many ways and I always ended up going back to printing an error and exiting manually for various reasons. But here it may be okay. I'll change it.

Reply:
[I decided](https://github.com/vmware/differential-datalog/pull/445/commits/cdaf2a6b01fd61c142f0a17baf136b92f3b90d58) to loop through a `Result` everywhere instead of dealing with `i32` values. I believe that makes the logic somewhat nicer.